mesonbuild/dependencies/base.py

```diff
@@ -424,6 +425,20 @@ def _set_libs(self):
         self.link_args = []
         libpaths = []
         for lib in shlex.split(out):
+            # MSVC cannot handle MinGW-esque /c/foo paths, convert them to C:/foo.
+            # We cannot resolve other paths starting with / like /home/foo so leave
+            # them as-is so the user gets an error/warning from the compiler/linker.
```

Comment:
There is an API for doing this path conversion, which you should probably use.

Further, it looks like you've hardcoded this conversion in a form which is only going to work for C:\

Reply:
Uh, oops. I originally had it as `str.isalph` but changed it to 'c' for testing.

What API are you talking about?

Reply:
(There is `cygpath -m /home/foo` or `cygpath -w /home/foo`)

Reply:
Relies on Cygwin being installed, presumably. This needs to work on MSVC-only setups.

Reply:
ah, ok.

Reply:
Um... so where are these paths coming from? msys2 .pc files?

Reply:
.pc files for libraries built with a MinGW toolchain, which can be MSYS or MSYS2. F.ex, the [gstreamer win32 binaries](https://gstreamer.freedesktop.org/data/pkg/windows/).

Actually, cross-mingw toolchain .pc files will also be broken in interesting ways but that's for another time I guess.